go/vt/vttablet/tabletmanager/rpc_external_reparent.go

```diff
@@ -180,15 +186,38 @@ func (agent *ActionAgent) finalizeTabletExternallyReparented(ctx context.Context
 
 			// We now know more about the old master, so add it to event data.
 			ev.OldMaster = *oldMasterTablet
+			tablets <- *oldMasterTablet
 		}()
 	}
 
-	tablet := agent.Tablet()
+	// update any other tablets claiming to be MASTER also to REPLICA
+	for alias, tabletInfo := range tabletMap {
+		if alias != topoproto.TabletAliasString(agent.TabletAlias) && alias != topoproto.TabletAliasString(oldMasterAlias) && tabletInfo.Tablet.Type == topodatapb.TabletType_MASTER {
+			log.Infof("finalizeTabletExternallyReparented: updating tablet record for another old master: %v", alias)
+			wg.Add(1)
+			go func() {
```

Comment:
Need to pass in a copy of any loop variables needed inside the goroutine, as you do below.